src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -7479,6 +7479,10 @@ void StorageReplicatedMergeTree::createTableSharedID() const
             id = zookeeper->get(zookeeper_table_id_path);
             LOG_DEBUG(log, "Shared ID on path {} concurrently created, will set ID {}", zookeeper_table_id_path, id);
         }
+        else if (code == Coordination::Error::ZNONODE)
+        {
+            LOG_DEBUG(log, "Shared ID on path {} cannot be created, table is completely dropped, will use {}", zookeeper_table_id_path, id);
+        }
```

Comment:
So `table_shared_id` will not be safe and `getTableSharedID` will remove Nil, is it ok? Of course we don't really care about dropped table, but it breaks invariant "getTableSharedID must return something meaningful"

Reply:
Yes, it's ok to return random trash from this function.

Reply:
Maybe remove this branch and move `createTableSharedID` from ctor to `startup`?